src/pages/home/search/SearchModal.tsx

```diff
@@ -0,0 +1,105 @@
+import { Dialog, Transition } from "@headlessui/react";
+import { useEffect, useState } from "react";
+import { BiSearch } from "react-icons/bi";
+
+import useLocalStorage from '../../../hooks/useLocalStorage'
+import styles from "./searchModal.module.css";
+
+// This component use headlessUi react library for modal
+function SearchModal() {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [savedSearch, setSavedSearch] = useLocalStorage("SWAD_SAVED_SEARCH", []);
+  
+  /**
+   * Function to check the operating system
+   * @returns boolean value if it is apple or not
+   */
+  const isAppleOS = () => {
+    const platform =
+      navigator?.userAgentData?.platform || navigator?.platform || "unknown";
+    return /(MAC|iPhone|iPod|iPad)/i.test(platform);
+  };
+
+  const keyBoardShortCut = isAppleOS() ? "⌘K" : "Ctrl+k";
+  const onKeyDown = (event: KeyboardEvent) => {
+    if (isOpen) return;
+    if ((event.metaKey || event.ctrlKey) && event.key === "k") {
+      event.preventDefault();
+      setIsOpen(true);
+    }
+  };
+
+   const onSearch =(event:React.ChangeEvent<HTMLInputElement>)=>{
+    const recentSearch = [...savedSearch];
+    // currently it is saving each word is typed, need to handle with debounce
+    recentSearch.push(event.target.value)
+    setSavedSearch(recentSearch)
+   }
+  useEffect(() => {
+    window.addEventListener("keydown", onKeyDown);
+
+    return () => {
+      window.removeEventListener("keydown", onKeyDown);
+    };
+  });
+
+  return (
+    <>
+      <button
+        type="button"
+        className={styles.bait_input}
+        aria-label="Open recipe search modal"
+        onClick={() => setIsOpen(true)}
+      >
+        <BiSearch size={24} aria-hidden />
+        <p className={styles.text}>Search recipes...</p>
```

Comment:
maybe a wrapper with `flex: 1; text-align: start;` might remove the excess space between search icon and search text? that way we don't need to hardcode negative margin?

Reply:
done